src/storage/conversion/TemplateEngine.ts

```diff
@@ -0,0 +1,44 @@
+import { promises as fsPromises } from 'fs';
+import { joinFilePath, resolveAssetPath } from '../../util/PathUtil';
+import Dict = NodeJS.Dict;
+
+export interface TemplateOptions {
+  // String contents of the template
+  template?: string;
+  // Name of the template file
+  templateFile?: string;
+  // Path of the template file
+  templatePath?: string;
+}
```

Comment:
Yeah I had it as above; Components.js was the problem. Is there a way to ignore, but also be able to use it in the params? Because that would be what I need.

Reply:
Afaik everything should work after adding the type name to the componentsignore file.

Reply:
I'll try. Just confused at how Components.js then can pick it up, if it doesn't know the property names.

Reply:
Components.js doesn't have to pick it up since it's not a constructor parameter. Should we ever have a constructor parameter that needs it we would have to add a new flag in the comment to indicate that it shouldn't worry about the typings of that field.

Reply:
> not a constructor parameter

It is, to `HandlebarsTemplateEngine`. How do I proceed?

Reply:
Ah I missed that. You can add the json `@range` like we do here: https://github.com/solid/community-server/blob/232a47d294b92ce539e74abb0020e9706e5527e2/src/identity/configuration/IdentityProviderFactory.ts#L70

That does mean components.js is not going to check if the fields in the config are valid, but I don't think there's a way to do that since we're using a union type. Even with that I do think it's better to have the union type for the code.

Unless @rubensworks knows of a better solution

Reply:
Great, thanks.

Reply:
For now, a `json` type may indeed be the best solution.
If union support is ever supported in Components.ks (not trivial), then we may reconsider.

Reply:
Thanks!

> (not trivial)

Personally, I wouldn't need the exclusive type-checking; just the union of allowed parameters. But yeah, I can expect there to be conflicts as well. Yay for JSON support.